addons/mail/controllers/discuss.py

```diff
@@ -305,19 +306,19 @@ def mail_attachment_upload(self, ufile, thread_id, thread_model, is_pending=Fals
     @http.route('/mail/attachment/delete', methods=['POST'], type='json', auth='public')
     def mail_attachment_delete(self, attachment_id, access_token=None, **kwargs):
         attachment_sudo = request.env['ir.attachment'].browse(int(attachment_id)).sudo().exists()
+        guest = request.env['mail.guest']._get_guest_from_request(request)
+        message_sudo = guest.env['mail.message'].sudo().search([('attachment_ids', 'in', attachment_sudo.ids)], limit=1)
```

Comment:
is the message guaranteed to exist/guaranteed to be found?
In chatter it's possible to have attachment linked to no message and those should be deletable as well.